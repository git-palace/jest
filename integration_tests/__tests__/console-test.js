/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 * @emails oncall+jsinfra
 */
'use strict';

const runJest = require('../runJest');
const skipOnWindows = require('skipOnWindows');
const {extractSummary} = require('../utils');

skipOnWindows.suite();

test('console printing', () => {
  const {stderr, status} = runJest('console');
  const {summary, rest} = extractSummary(stderr);

  expect(status).toBe(0);
  expect(rest).toMatchSnapshot();
  expect(summary).toMatchSnapshot();
});

test('console printing with --verbose', () => {
  const {stderr, stdout, status} =
    runJest('console', ['--verbose', '--no-cache']);
  const {summary, rest} = extractSummary(stderr);

  expect(status).toBe(0);
  expect(stdout).toMatchSnapshot();
  expect(rest).toMatchSnapshot();
  expect(summary).toMatchSnapshot();
});
