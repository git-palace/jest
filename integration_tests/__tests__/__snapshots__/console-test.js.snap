exports[`test console printing 1`] = `
" PASS  __tests__/console-test.js
  ● Console

    console.log __tests__/console-test.js:11
      This is a log message.
    console.info __tests__/console-test.js:13
      This is an info message.
    console.warn __tests__/console-test.js:15
      This is a warning message.
    console.error __tests__/console-test.js:17
      This is a error message.

 PASS  __tests__/console2-test.js
  ● Console

    console.error __tests__/console2-test.js:18
      This is an error from another test file.
"
`;

exports[`test console printing with --verbose 1`] = `
"  console.log __tests__/console-test.js:11
    This is a log message.

  console.info __tests__/console-test.js:13
    This is an info message.

  console.warn __tests__/console-test.js:15
    This is a warning message.

  console.error __tests__/console-test.js:17
    This is a error message.

  console.error __tests__/console2-test.js:18
    This is an error from another test file.

"
`;
